process/block/displayBlock.go

```diff
@@ -56,16 +68,63 @@ func (txc *transactionCounter) getPoolCounts(poolsHolder dataRetriever.PoolsHold
 	return
 }
 
-// subtractRestoredTxs updated the total processed txs in case of restore
-func (txc *transactionCounter) subtractRestoredTxs(txsNr int) {
+// headerReverted updates the total processed txs in case of restore. It also sets the current block txs to 0
+func (txc *transactionCounter) headerReverted(hdr data.HeaderHandler) {
+	if check.IfNil(hdr) {
+		log.Warn("programming error: nil header in transactionCounter.headerReverted function")
+		return
+	}
+
+	currentBlockTxs := txc.getProcessedTxCount(hdr)
+
 	txc.mutex.Lock()
-	defer txc.mutex.Unlock()
-	if txc.totalTxs < uint64(txsNr) {
+	txc.currentBlockTxs = 0
+	txc.safeSubtractTotalTxs(uint64(currentBlockTxs))
+	txc.appStatusHandler.SetUInt64Value(common.MetricNumProcessedTxs, txc.totalTxs)
+	txc.mutex.Unlock()
+}
+
+func (txc *transactionCounter) safeSubtractTotalTxs(delta uint64) {
+	if txc.totalTxs < delta {
 		txc.totalTxs = 0
 		return
 	}
 
-	txc.totalTxs -= uint64(txsNr)
+	txc.totalTxs -= delta
+}
+
+func (txc *transactionCounter) headerExecuted(hdr data.HeaderHandler) {
+	if check.IfNil(hdr) {
+		log.Warn("programming error: nil header in transactionCounter.headerExecuted function")
+		return
+	}
+
+	currentBlockTxs := txc.getProcessedTxCount(hdr)
+
+	txc.mutex.Lock()
+	txc.currentBlockTxs = uint64(currentBlockTxs)
+	txc.totalTxs += uint64(currentBlockTxs)
+	txc.appStatusHandler.SetUInt64Value(common.MetricNumProcessedTxs, txc.totalTxs)
+	txc.mutex.Unlock()
+}
+
+func (txc *transactionCounter) getProcessedTxCount(hdr data.HeaderHandler) int32 {
+	currentBlockTxs := int32(0)
+	for _, miniBlockHeaderHandler := range hdr.GetMiniBlockHeaderHandlers() {
+		if miniBlockHeaderHandler.GetTypeInt32() == int32(block.PeerBlock) {
```

Comment:
So only TxBlock, SmartContractResultBlock and RewardsBlock mb types will matter for this computation?

Reply:
yes. To be decided.